lib/android/app/src/main/java/com/reactnativenavigation/react/coordinator/CoordinatorLayoutView.java

```diff
@@ -0,0 +1,54 @@
+package com.reactnativenavigation.react.coordinator;
+
+
+import android.content.Context;
+import android.support.design.widget.AppBarLayout;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.v4.widget.NestedScrollView;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+
+public class CoordinatorLayoutView extends CoordinatorLayout {
+
+	public CoordinatorLayoutView(Context context) {
+		super(context);
+		init();
+	}
+
+	public CoordinatorLayoutView(Context context, AttributeSet attrs) {
+		super(context, attrs);
+		init();
+	}
+
+	public CoordinatorLayoutView(Context context, AttributeSet attrs, int defStyleAttr) {
+		super(context, attrs, defStyleAttr);
+		init();
+	}
+
+	private void init() {
+		LayoutParams params =
+				new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+		params.setBehavior(new AppBarLayout.ScrollingViewBehavior());
+		this.setLayoutParams(params);
+		this.setFitsSystemWindows(false);
+	}
+
+	private final Runnable measureAndLayout = new Runnable() {
+		@Override
+		public void run() {
+			measure(
+					MeasureSpec.makeMeasureSpec(getWidth(), MeasureSpec.EXACTLY),
+					MeasureSpec.makeMeasureSpec(getHeight(), MeasureSpec.EXACTLY));
+			layout(getLeft(), getTop(), getRight(), getBottom());
+		}
+	};
+
+	@Override
+	public void requestLayout() {
+		super.requestLayout();
+		post(measureAndLayout);
```

Comment:
why is this needed? you already set MATCH_PARENT on yourself

Reply:
It's some kind of bug from RN. It doesn't measure children properly without this 'hack'

Reply:
hmm. it's worth a comment then, explaining that. maybe RN50 solves it?..